rastervision_core/rastervision/core/utils/stac.py

```diff
@@ -28,32 +32,73 @@ def my_write_method(uri, txt):
         else:
             STAC_IO.default_write_text_method(uri, txt)
 
-    STAC_IO.read_text_method = my_read_method
-    STAC_IO.write_text_method = my_write_method
+    STAC_IO.read_text_method = read_method
+    STAC_IO.write_text_method = write_method
+
+
+def parse_stac(stac_uri: str) -> dict:
+    """Parse a STAC catalog JSON file to extract label URI, images URIs,
+    and AOIs.
 
+    Args:
+        stac_uri (str): Path to the STAC catalog JSON file.
 
-def parse_stac(stac_uri):
-    setup_stac_s3()
+    Returns:
+        dict: A dict with keys: "labels_uri", "labels_bbox", "image_uris",
+        and "aoi_geometry".
+    """
+    setup_stac_io()
     cat = Catalog.from_file(stac_uri)
     cat.make_all_asset_hrefs_absolute()
-    labels_uri = None
-    geotiff_uris = []
-    for item in cat.get_all_items():
-        if isinstance(item, LabelItem):
-            labels_uri = list(item.assets.values())[0].href
-            labels_box = box(*item.bbox)
-
-    # only use geotiffs that intersect with bbox of labels
-    for item in cat.get_all_items():
-        if not isinstance(item, LabelItem):
-            geotiff_uri = list(item.assets.values())[0].href
-            geotiff_box = box(*item.bbox)
-            if labels_box.intersects(geotiff_box):
-                geotiff_uri = geotiff_uri.replace('%7C', '|')
-                geotiff_uris.append(geotiff_uri)
-
-    if not labels_uri:
-        raise ValueError('Unable to read labels URI from STAC.')
-    if not geotiff_uris:
-        raise ValueError('Unable to read GeoTIFF URIs from STAC.')
-    return labels_uri, labels_box, geotiff_uris
+
+    label_items = [
+        item for item in cat.get_all_items() if 'label' in item.stac_extensions
+    ]
```

Comment:
This should use the `label` extension URI, since these shorthands are formally speaking not allowed anymore. I'll update the STAC exports in RF to do the same.

That url is at: https://raw.githubusercontent.com/stac-extensions/label/v1.0.0/json-schema/schema.json

That was changed in [1.0.0-rc.4](https://github.com/radiantearth/stac-spec/blob/master/CHANGELOG.md#v100-rc4---2021-05-11)

Reply:
You mean that label items will have a `uri` field containing the URL `"https://raw.githubusercontent.com/stac-extensions/label/v1.0.0/json-schema/schema.json"`?

Reply:
That's correct, or, you know, they will after raster-foundry/raster-foundry#5614

Reply:
Another note -- yesterday I released a bugfix on the label extension, so there's now also a [v1.0.1](https://stac-extensions.github.io/label/v1.0.1/schema.json). Also, these can be addressed either by the `stac-extensions.github.io` address or the `raw.githubusercontent.com` address, since the STAC rules at this point are to resolve the schema and validate rather than convert the shorthand (I don't think consumers who want to identify items of a certain type were a big part of that conversation, though I wasn't really in it so I can't say that for certain).

Reply:
Is resolving the schema and checking that title == "Label Extension" good enough?

Reply:
Oh that's a smart way to do it 👍🏻